go/cli/internal/cli/deploy.go

```diff
@@ -0,0 +1,296 @@
+	k8sClient, err := createKubernetesClient()
+	if err != nil {
+		return fmt.Errorf("failed to create Kubernetes client: %v", err)
+	}
+
+	// Handle secret creation or reference
+	var secretName string
+	if cfg.APIKeySecret != "" {
+		// Use existing secret
+		secretName = cfg.APIKeySecret
+		// Verify the secret exists
+		if err := verifySecretExists(k8sClient, cfg.Config.Namespace, secretName, apiKeyEnvVar); err != nil {
+			return fmt.Errorf("failed to verify secret '%s': %v", secretName, err)
+		}
+		fmt.Printf("Using existing secret '%s' in namespace '%s'\n", secretName, cfg.Config.Namespace)
+	} else if cfg.APIKey != "" {
+		// Create new secret with provided API key
+		secretName = fmt.Sprintf("%s-%s", manifest.Name, strings.ToLower(manifest.ModelProvider))
+		if err := createSecret(k8sClient, cfg.Config.Namespace, secretName, apiKeyEnvVar, cfg.APIKey); err != nil {
+			return fmt.Errorf("failed to create secret: %v", err)
+		}
+	} else {
+		return fmt.Errorf("either --api-key or --api-key-secret must be provided")
+	}
+
+	// Create the Agent CRD
+	if err := createAgentCRD(k8sClient, cfg, manifest, secretName, apiKeyEnvVar); err != nil {
+		return fmt.Errorf("failed to create Agent CRD: %v", err)
+	}
+
+	fmt.Printf("Successfully deployed agent '%s' to namespace '%s'\n", manifest.Name, cfg.Config.Namespace)
+	return nil
+}
+
+// loadManifest loads the kagent.yaml file from the project directory
+func loadManifest(projectDir string) (*common.AgentManifest, error) {
+	// Use the Manager to load the manifest
+	manager := common.NewManifestManager(projectDir)
+	manifest, err := manager.Load()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load kagent.yaml: %v", err)
+	}
+
+	// Additional validation for deploy-specific requirements
+	if manifest.ModelProvider == "" {
+		return nil, fmt.Errorf("model provider is required in kagent.yaml")
+	}
+
+	return manifest, nil
+}
+
+// getAPIKeyEnvVar returns the environment variable name for the given model provider
+func getAPIKeyEnvVar(modelProvider string) string {
+	switch modelProvider {
+	case string(v1alpha2.ModelProviderAnthropic):
+		return "ANTHROPIC_API_KEY"
+	case string(v1alpha2.ModelProviderOpenAI):
+		return "OPENAI_API_KEY"
+	case string(v1alpha2.ModelProviderAzureOpenAI):
+		return "AZURE_API_KEY"
+	case string(v1alpha2.ModelProviderGemini):
+		return "GOOGLE_API_KEY"
+	default:
+		return ""
+	}
+}
+
+// createKubernetesClient creates a Kubernetes client
+func createKubernetesClient() (client.Client, error) {
+	// Try to load from kubeconfig first
+	kubeconfig := os.Getenv("KUBECONFIG")
+	if kubeconfig == "" {
+		home := os.Getenv("HOME")
+		if home != "" {
+			kubeconfig = filepath.Join(home, ".kube", "config")
+		}
+	}
+
+	var config *rest.Config
+	var err error
+
+	if kubeconfig != "" && fileExists(kubeconfig) {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		// Try in-cluster config
+		config, err = rest.InClusterConfig()
```

Comment:
Why are you trying this? this should never be running in cluster

Reply:
this was a mistake I updated this to use `clientcmd.NewNonInteractiveDeferredLoadingClientConfig` instead which is a cleaner way to load the kubeConfig.